### PR TITLE
[4.0] plugin installation order

### PR DESCRIPTION
**Pull request by brianteeman**
This PR does two things

The first is simply to put the list of plugins to be installed back into alphabetical order. This makes it easier to maintain and find things in the database

The second is to set the plugin order

In general it probably doesn't matter at all what order the plugins are in but if you look at the ordering dropdown it looks a mess as many of the plugins have the same order number. Some of the oldest plugins have an ordering value but most just have a 0.

This PR adds a value starting at 1 for each plugin grouped by type

This is only for new installs
This is only for mysql (if approved I will update postgresql as well)

### Example Before
image

### Example After
image

### CODE REVIEW

**Comment by brianteeman** [comment]
Can someone restart drone please

**Comment by HLeithner** [comment]
Restarting drone doesn't help. RIPS: `Failure: Number of issues exceeds sum threshold (3/0)` @SniperSister @zero-24 can you have a look at RIPS thanks

**Comment by zero-24** [comment]
Has this PR been updated to the latest 4.0-dev? Just restarting does not help with rips as it tests the same codebase.

**Comment by brianteeman** [comment]
It was the latest at the time it was submitted. I will update it again

**Comment by shinderohanoffical** [comment]
I have tested this item :white_check_mark: successfully on 984fe0eb75ad00b24e6312fbdb894e1efafbe4e1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31489">issues.joomla.org/tracker/joomla-cms/31489</a>.</sub>

**Comment by snehal0904** [comment]
I have tested this item :white_check_mark: successfully on 984fe0eb75ad00b24e6312fbdb894e1efafbe4e1<br /><br />Tested successfully.

Thank you for the fix<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31489">issues.joomla.org/tracker/joomla-cms/31489</a>.</sub>

**Comment by richard67** [comment]
> This is only for mysql (if approved I will update postgresql as well)

Waiting with RTC since postgresql files are missing.

**Comment by brianteeman** [comment]
@richard67 as stated in the first post I am waiting to hear if its approved before spending the time to do the postgres version

**Comment by brianteeman** [comment]
@richard67 updated with postgres

File: installation/sql/postgresql/base.sql [hunk 1]
```diff
@@ -244,131 +244,131 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
...
+(0, 'plg_user_contactcreator', 'plugin', 'contactcreator', 'user', 0, 0, 1, 0, 1, '', '{"autowebpage":"","category":"4","autopublish":"0"}', '', 1, 0),
+(0, 'plg_user_joomla', 'plugin', 'joomla', 'user', 0, 1, 1, 0, 1, '', '{"autoregister":"1","mail_to_user":"1","forceLogout":"1"}', '', 2, 0),
+(0, 'plg_user_profile', 'plugin', 'profile', 'user', 0, 0, 1, 0, 1, '', '{"register-require_address1":"1","register-require_address2":"1","register-require_city":"1","register-require_region":"1","register-require_country":"1","register-require_postal_code":"1","register-require_phone":"1","register-require_website":"1","register-require_favoritebook":"1","register-require_aboutme":"1","register-require_tos":"1","register-require_dob":"1","profile-require_address1":"1","profile-require_address2":"1","profile-require_city":"1","profile-require_region":"1","profile-require_country":"1","profile-require_postal_code":"1","profile-require_phone":"1","profile-require_website":"1","profile-require_favoritebook":"1","profile-require_aboutme":"1","profile-require_tos":"1","profile-require_dob":"1"}', '', 1, 0),
```
Comment (richard67) [inline installation/sql/postgresql/base.sql]:
```suggestion
(0, 'plg_user_profile', 'plugin', 'profile', 'user', 0, 0, 1, 0, 1, '', '{"register-require_address1":"1","register-require_address2":"1","register-require_city":"1","register-require_region":"1","register-require_country":"1","register-require_postal_code":"1","register-require_phone":"1","register-require_website":"1","register-require_favoritebook":"1","register-require_aboutme":"1","register-require_tos":"1","register-require_dob":"1","profile-require_address1":"1","profile-require_address2":"1","profile-require_city":"1","profile-require_region":"1","profile-require_country":"1","profile-require_postal_code":"1","profile-require_phone":"1","profile-require_website":"1","profile-require_favoritebook":"1","profile-require_aboutme":"1","profile-require_tos":"1","profile-require_dob":"1"}', '', 3, 0),
```

File: installation/sql/postgresql/base.sql [hunk 2]
```diff
@@ -244,131 +244,131 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
...
+(0, 'plg_user_terms', 'plugin', 'terms', 'user', 0, 0, 1, 0, 1, '', '{}', '', 2, 0),
```
Comment (richard67) [inline installation/sql/postgresql/base.sql]:
```suggestion
(0, 'plg_user_terms', 'plugin', 'terms', 'user', 0, 0, 1, 0, 1, '', '{}', '', 4, 0),
```

File: installation/sql/postgresql/base.sql [hunk 3]
```diff
@@ -244,131 +244,131 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
...
+(0, 'plg_user_token', 'plugin', 'token', 'user', 0, 1, 1, 0, 1, '', '{}', '', 3, 0),
```
Comment (richard67) [inline installation/sql/postgresql/base.sql]:
```suggestion
(0, 'plg_user_token', 'plugin', 'token', 'user', 0, 1, 1, 0, 1, '', '{}', '', 5, 0),
```

**Review by richard67 — COMMENTED** [review]
Found some differences in ordering between postgresql and mysql.

**Comment by richard67** [comment]
We have 2 good tests with MySQL (or MariaDB), and I've reviewed the PR and can confirm that it does what it claims to do and that it's the same for MySQL and PostgreSQL beside the necessary difference in names quoting.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31489">issues.joomla.org/tracker/joomla-cms/31489</a>.</sub>

**Comment by brianteeman** [comment]
Any reason why this RTC pr has not been merged?

**Comment by HLeithner** [comment]
I'm not so sure that the order has no impact, I know that the `languagefilter` plugins could produce troubles if it's not the loaded before some other system plugins but can only name my own plugin.

**Comment by brianteeman** [comment]
Only one core type of plugins are impacted by the order. The installation plugins and that is why they are not ordered alphabetically in this PR.

There is an impact of the order with 3pd plugins. Some of them absolutely do need to be positioned before or after other plugins.  This PR actually helps that as can be seen in the screenshots.

Also as soon as you move any plugin (before this PR) they are re-ordered anyway - proving that there is no impact

**Comment by infograf768** [comment]
The alphaorder set here works for en-GB, not for other languages.
I get for French (some are en-GB below as they do not exist in 3.9.23 French pack)

Before patch:
Screen Shot 2020-12-16 at 10 13 46

after patch:
Screen Shot 2020-12-16 at 10 26 59

**Comment by brianteeman** [comment]
The point of the PR is not the order but the fact that they all have a unique number.

**Comment by wilsonge** [comment]
Consistent ordering seems like a good idea to me. Thanks!

**Comment by brianteeman** [comment]
thx

**Pull request merged**